com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java

```diff
@@ -0,0 +1,202 @@
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        LOGGER.info("Cleanup Job started by application instance: " + getInstanceIndex() + " at: " + Instant.now().toString());
+
+        Date expirationTime = getExpirationTime();
+
+        abortOldOperationsInStateError(expirationTime);
+
+        cleanUpFinishedOperationsData(expirationTime);
+
+        removeActivitiHistoricData(expirationTime);
+
+        LOGGER.info("Cleanup Job finished at: " + Instant.now().toString());
+    }
+
+    private String getInstanceIndex() {
+        Environment env = new Environment();
+        return env.getVariable("INSTANCE_INDEX");
+    }
+
+    private Date getExpirationTime() {
+        long maxTtlForOldData = configuration.getMaxTtlForOldData();
+        Date cleanUpTimestamp = Date.from(Instant.now().minusSeconds(maxTtlForOldData));
+        LOGGER.info("Will perform clean up for data stored before : " + cleanUpTimestamp.toString());
+        return cleanUpTimestamp;
+    }
+
+    private void abortOldOperationsInStateError(Date expirationTime) {
+        LOGGER.info("Aborting operations started before: " + expirationTime.toString());
+        List<Operation> activeOperationsInStateError = getActiveOperationsInStateError(expirationTime);
+        List<String> operationsInErrorIds = getProcessIds(activeOperationsInStateError);
+        executeAbortOperationAction(operationsInErrorIds);
+        LOGGER.info("Aborted operations count : " + operationsInErrorIds.size());
+    }
+
+    private void cleanUpFinishedOperationsData(Date expirationTime) {
+        LOGGER.info("Cleaning up data for finished operations started before: " + expirationTime.toString());
+        List<Operation> finishedOperations = getFinishedOperations(expirationTime);
+        List<String> finishedProcessIds = getProcessIds(finishedOperations);
+        LOGGER.debug("Data will be cleaned up for operations: " + finishedProcessIds);
+
+        removeProgressMessages(finishedProcessIds);
+        removeProcessLogs(finishedProcessIds);
+
+        Map<String, String> processIdToAppArchiveId = getProcessIdToAppArchiveId();
+
+        Map<String, List<String>> spaceToProcessIds = getSpaceToProcessIds(finishedOperations);
+        Map<String, List<String>> spaceToFileIds = getSpaceToFileIds(spaceToProcessIds, processIdToAppArchiveId);
+
+        removeOldFilesFromDb(spaceToFileIds);
+        removeOldFilesFromFs(spaceToFileIds);
+
+    }
+
+    private void removeActivitiHistoricData(Date expirationTime) {
+        List<HistoricProcessInstance> historicProcess = activitiFacade.getHistoricProcessInstancesFinishedAndStartedBefore(expirationTime);
+        if (CollectionUtils.isEmpty(historicProcess)) {
+            return;
+        }
+        for (HistoricProcessInstance historicProcessInstance : historicProcess) {
+            if (historicProcessInstance != null && historicProcessInstance.getEndTime() != null) {
```

Comment:
Could the historicProcessInstance be null? Consider switching to Java 8 stream API to filter and iterate.

Reply:
Why check for historicProcessInstance.getEndTime() != null? Here we should just remove all old data.

Reply:
Done.